src/main/io/gps.c

```diff
@@ -1540,132 +1712,145 @@ static bool UBLOX_parse_gps(void)
-        next_fix = (_buffer.solution.fix_status & NAV_STATUS_FIX_VALID) && (_buffer.solution.fix_type == FIX_3D);
-        if (!next_fix)
-            DISABLE_STATE(GPS_FIX);
-        gpsSol.numSat = _buffer.solution.satellites;
+        case MSG_NAV_POSLLH:
+            *gpsPacketLogChar = LOG_UBLOX_POSLLH;
+            //i2c_dataset.time                = _buffer.posllh.time;
+            gpsSol.llh.lon = _buffer.posllh.longitude;
+            gpsSol.llh.lat = _buffer.posllh.latitude;
+            gpsSol.llh.altCm = _buffer.posllh.altitudeMslMm / 10;  //alt in cm
+            gpsSetFixState(next_fix);
+            _new_position = true;
+            break;
+        case MSG_NAV_STATUS:
+            *gpsPacketLogChar = LOG_UBLOX_STATUS;
+            next_fix = (_buffer.status.fix_status & NAV_STATUS_FIX_VALID) && (_buffer.status.fix_type == FIX_3D);
+            if (!next_fix)
+                DISABLE_STATE(GPS_FIX);
+            break;
+        //case MSG_MON_VER:
+        case MSG_NAV_DOP:
+            if (_class == CLASS_MON) {
+                memcpy(&gpsData.monVer, &_buffer.ver.swVersion, 40 + 10 * 30);
+                gpsData.acquiredMonVer = true;
+                gpsData.unitVersion = ubloxDetectVersion(gpsData.monVer.hwVersion, sizeof(gpsData.monVer.hwVersion));
+                gpsData.ackState = UBLOX_ACK_GOT_ACK;
+            } else {
+                *gpsPacketLogChar = LOG_UBLOX_DOP;
+                gpsSol.dop.pdop = _buffer.dop.pdop;
+                gpsSol.dop.hdop = _buffer.dop.hdop;
+                gpsSol.dop.vdop = _buffer.dop.vdop;
+            }
+            break;
+        case MSG_NAV_SOL:
+            *gpsPacketLogChar = LOG_UBLOX_SOL;
+            next_fix =
+                    (_buffer.solution.fix_status & NAV_STATUS_FIX_VALID) && (_buffer.solution.fix_type == FIX_3D);
+            if (!next_fix)
+                DISABLE_STATE(GPS_FIX);
+            gpsSol.numSat = _buffer.solution.satellites;
 #ifdef USE_RTC_TIME
-        //set clock, when gps time is available
-        if(!rtcHasTime() && (_buffer.solution.fix_status & NAV_STATUS_TIME_SECOND_VALID) && (_buffer.solution.fix_status & NAV_STATUS_TIME_WEEK_VALID)) {
-            //calculate rtctime: week number * ms in a week + ms of week + fractions of second + offset to UNIX reference year - 18 leap seconds
-            rtcTime_t temp_time = (((int64_t) _buffer.solution.week)*7*24*60*60*1000) + _buffer.solution.time + (_buffer.solution.time_nsec/1000000) + 315964800000LL - 18000;
-            rtcSet(&temp_time);
-        }
+            //set clock, when gps time is available
+            if (!rtcHasTime() && (_buffer.solution.fix_status & NAV_STATUS_TIME_SECOND_VALID) &&
+                (_buffer.solution.fix_status & NAV_STATUS_TIME_WEEK_VALID)) {
+                //calculate rtctime: week number * ms in a week + ms of week + fractions of second + offset to UNIX reference year - 18 leap seconds
+                rtcTime_t temp_time =
```

Comment:
Formatting changes make it really hard to find the changes that actually address why M10s weren't working.
Perhaps leave the formatting alone until the core changes are done, and then maybe address those as a last cleanup commit (if at all in this PR)?

Reply:
Agree fix and refactoring should be separated.

Reply:
The code seems OK, and no better alternative is provided, so I'll accept the comments generally, and resolve the conversation.